python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -183,6 +183,82 @@ def __init__(self, experiment_checkpoint_path, trials=None):
         super(ExperimentAnalysis, self).__init__(
             os.path.dirname(experiment_checkpoint_path))
 
+    def get_best_trial(self, metric, mode="max", scope="all"):
+        """Retrieve the best trial object. Compare all trials' scores on
+            `metric`, based on `mode`. Only factor in `scope` steps.
+
+        Args:
+            metric (str): Key for trial info to order on.
+            mode (str): One of [min, max].
+            scope (str): One of [all, last].
+        """
+        if mode not in ["max", "min"]:
+            logger.warning(
```

Comment:
hmm should we `raise ValueError` or just `log`? might be due to a bug in the user's logic that needs to be surfaced.